docs/src/guide/queries_iteration.md

```diff
@@ -0,0 +1,292 @@
++++
+type = "docs"
+title = "Queries and iteration"
+weight = 40
++++
+
+Iterating over entities can be done via 
+classic `for` loops applied to [queries](#queries),
+or via an [`apply`](#applying-functions-to-entities-in-queries) 
+operation, which applies a given function to all entities 
+conforming to a query.
+
+## Queries
+
+{{< api Query Queries >}} allow to iterate over all 
+entities with or without a specific 
+set of components. To create a query, we can use 
+the {{< api World.query query >}} method of 
+{{< api World >}}. The parameters used in this method are the components
+that each entity we look for must have. For example, if we want to
+iterate over all entities with a `Position` and a `Velocity` component,
+we can do this as follows:
+
+```mojo {doctest="guide_queries_iteration" global=true hide=true}
+from larecs import World, MutableEntityAccessor
+
+@value
+struct Position:
+    var x: Float64
+    var y: Float64
+
+@value
+struct Velocity:
+    var dx: Float64
+    var dy: Float64
+```
+
+```mojo {doctest="guide_queries_iteration" hide=true}
+world = World[Position, Velocity]()
+```
+
+```mojo {doctest="guide_queries_iteration"}
+# Add entities with different components
+_ = world.add_entity()
+_ = world.add_entity(Position(0, 0))
+_ = world.add_entity(Velocity(1, 0))
+_ = world.add_entity(Position(1, 0), Velocity(1, 0))
+
+# Query all entities that have a position
+query = world.query[Position]()
+
+# Of the entities we have just added,
+# two have a position component
+print(len(query)) # "2"
+
+# Now let us iterate over the queried entities
+for entity in query:
+    pos = entity.get_ptr[Position]()
+    print(
+        "Entity at position: (" 
+        + String(pos[].x) + ", " + String(pos[].y) + ")"
+    )
+```
+
+Queries can be adjusted to also exclude entities that have
+certain components. For example, if we want to iterate
+over all entities that have a `Position` component
+but not a `Velocity` component, we can do this 
+using the {{< api Query.without without >}} method:
+
+```mojo {doctest="guide_queries_iteration"}
+excluding_query = world.query[Position]().without[Velocity]()
+print(len(excluding_query)) # "1"
+```
+
+Furthermore, we can also query for entities that have
+exactly the components we are looking for but no more.
+This can be done using the {{< api Query.exclusive exclusive >}} 
+method. For example, if we want to iterate
+over all entities that have only a `Position` component,
+we can do this as follows:
+
+```mojo {doctest="guide_queries_iteration"}
+excluding_query = world.query[Position]().exclusive()
+print(len(excluding_query)) # "1"
+```
+
+> [!Note] 
+> Determining the length of a query is not a trivial operation
+> and can be expensive if the ECS is involves many components.
+> Therefore, it is advisable to avoid applying the `len` function
+> on queries in "hot" code.
```

Comment:
In the Ark user guide, I mentioned that it is still way faster than using iteration for counting.